test/unit/v2/database.test.ts

```diff
@@ -265,4 +264,53 @@ describe("database service", () => {
       expect(true).toBeTruthy();
     }
   });
+
+  it("deprecated option throws error", async () => {
+    server.use(
+      // Query against system engine
+      rest.post(
+        `https://some_system_engine.com/${QUERY_URL}`,
+        (req, res, ctx) => {
+          const body = (String(req.body) ?? "").toLowerCase();
+          if (
```

Comment:
I'd  re-write this snippet as following:

```
  if (body.includes("information_schema.engines")) {
    if (body.includes("some_engine")) {
        return res(ctx.json(selectEngineResponse));
    } else if (body.includes("some_other_engine")) {
        return res(ctx.json(selectOtherEngineResponse));
    } else {
        return res(ctx.json(selectEnginesResponse));
    }
  } else {
    return res(ctx.json(selectDbResponse));
  }

```

IMHO it is much more readable. Also then I'd extract the logic to separate function that returns response according to the body and then use it as following:

```
return res(ctx.json(getResponse(body)))';
```

Reply:
I don't think nested ifs are much more readable either. However, this test is much simpler - the setup in question was copied over from the previous test. We only need one case in this test so no nesting is necessary. I've removed redundant lines.